platform/iphone/gl_view.mm

```diff
@@ -379,369 +222,106 @@ - (void)destroyFramebuffer {
-		};
-	};
-}
-
-- (void)touchesEnded:(NSSet *)touches withEvent:(UIEvent *)event {
-	NSArray *tlist = [[event allTouches] allObjects];
-	for (unsigned int i = 0; i < [tlist count]; i++) {
-
-		if ([touches containsObject:[tlist objectAtIndex:i]]) {
-
-			UITouch *touch = [tlist objectAtIndex:i];
-			if (touch.phase != UITouchPhaseEnded)
-				continue;
-			int tid = get_touch_id(touch);
-			ERR_FAIL_COND(tid == -1);
-			remove_touch(touch);
-			CGPoint touchPoint = [touch locationInView:self];
-			OSIPhone::get_singleton()->touch_press(tid, touchPoint.x * self.contentScaleFactor, touchPoint.y * self.contentScaleFactor, false, false);
-		};
-	};
-}
-
-- (void)touchesCancelled:(NSSet *)touches withEvent:(UIEvent *)event {
-
-	OSIPhone::get_singleton()->touches_cancelled();
-	clear_touches();
-};
-
-- (BOOL)canBecomeFirstResponder {
-	return YES;
-};
-
-- (void)open_keyboard {
-	//keyboard_text = p_existing;
-	[self becomeFirstResponder];
-};
-
-- (void)hide_keyboard {
-	//keyboard_text = p_existing;
-	[self resignFirstResponder];
-};
-
-- (void)keyboardOnScreen:(NSNotification *)notification {
-	NSDictionary *info = notification.userInfo;
-	NSValue *value = info[UIKeyboardFrameEndUserInfoKey];
-
-	CGRect rawFrame = [value CGRectValue];
-	CGRect keyboardFrame = [self convertRect:rawFrame fromView:nil];
-
-	OSIPhone::get_singleton()->set_virtual_keyboard_height(_points_to_pixels(keyboardFrame.size.height));
-}
-
-- (void)keyboardHidden:(NSNotification *)notification {
-	OSIPhone::get_singleton()->set_virtual_keyboard_height(0);
-}
-
-- (void)deleteBackward {
-	if (keyboard_text.length())
-		keyboard_text.erase(keyboard_text.length() - 1, 1);
-	OSIPhone::get_singleton()->key(KEY_BACKSPACE, true);
-};
-
-- (BOOL)hasText {
-	return keyboard_text.length() ? YES : NO;
-};
-
-- (void)insertText:(NSString *)p_text {
-	String character;
-	character.parse_utf8([p_text UTF8String]);
-	keyboard_text = keyboard_text + character;
-	OSIPhone::get_singleton()->key(character[0] == 10 ? KEY_ENTER : character[0], true);
-	printf("inserting text with character %i\n", character[0]);
-};
-
-- (void)audioRouteChangeListenerCallback:(NSNotification *)notification {
-	printf("*********** route changed!\n");
-	NSDictionary *interuptionDict = notification.userInfo;
-
-	NSInteger routeChangeReason = [[interuptionDict valueForKey:AVAudioSessionRouteChangeReasonKey] integerValue];
-
-	switch (routeChangeReason) {
-
-		case AVAudioSessionRouteChangeReasonNewDeviceAvailable: {
-			NSLog(@"AVAudioSessionRouteChangeReasonNewDeviceAvailable");
-			NSLog(@"Headphone/Line plugged in");
-		}; break;
-
-		case AVAudioSessionRouteChangeReasonOldDeviceUnavailable: {
-			NSLog(@"AVAudioSessionRouteChangeReasonOldDeviceUnavailable");
-			NSLog(@"Headphone/Line was pulled. Resuming video play....");
-			if (_is_video_playing()) {
-
-				dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 0.5f * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
```

Comment:
Not sure why this was needed by a random `dispatch_after` to work around something is generally a sign of code smell. 👃